logstreams/src/test/java/io/zeebe/logstreams/state/NotReplicateAllChunksTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.logstreams.state;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.db.impl.DefaultColumnFamily;
+import io.zeebe.db.impl.rocksdb.ZeebeRocksDbFactory;
+import io.zeebe.logstreams.processor.SnapshotChunk;
+import io.zeebe.logstreams.processor.SnapshotReplication;
+import io.zeebe.test.util.AutoCloseableRule;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class NotReplicateAllChunksTest {
+
+  @Rule public TemporaryFolder tempFolderRule = new TemporaryFolder();
+  @Rule public AutoCloseableRule autoCloseableRule = new AutoCloseableRule();
+
+  private StateSnapshotController replicatorSnapshotController;
+  private StateSnapshotController receiverSnapshotController;
+  private EvilReplicator replicator;
+  private StateStorage receiverStorage;
+
+  @Before
+  public void setup() throws IOException {
+    final File runtimeDirectory = tempFolderRule.newFolder("runtime");
+    final File snapshotsDirectory = tempFolderRule.newFolder("snapshots");
+    final StateStorage storage = new StateStorage(runtimeDirectory, snapshotsDirectory);
+
+    final File receiverRuntimeDirectory = tempFolderRule.newFolder("runtime-receiver");
+    final File receiverSnapshotsDirectory = tempFolderRule.newFolder("snapshots-receiver");
+    receiverStorage = new StateStorage(receiverRuntimeDirectory, receiverSnapshotsDirectory);
+
+    replicator = new EvilReplicator();
+    replicatorSnapshotController =
+        new StateSnapshotController(
+            ZeebeRocksDbFactory.newFactory(DefaultColumnFamily.class), storage, replicator, 1);
+    receiverSnapshotController =
+        new StateSnapshotController(
+            ZeebeRocksDbFactory.newFactory(DefaultColumnFamily.class),
+            receiverStorage,
+            replicator,
+            1);
+
+    autoCloseableRule.manage(replicatorSnapshotController);
+    autoCloseableRule.manage(receiverSnapshotController);
+    replicatorSnapshotController.openDb();
```

Comment:
The setup is similar to `InvalidChecksumSnapshotChunkTest`. Maybe, we can merge both tests together. Just move the test method and the replicator.